sdk/storage/azure-storage-file-share/azure/storage/fileshare/_share_client.py

```diff
@@ -360,6 +360,33 @@ def delete_share(
         except StorageErrorException as error:
             process_storage_error(error)
 
+    @distributed_trace
+    def undelete_share(self, deleted_share_name, deleted_share_version, **kwargs):
+        # type: (str, str, **Any) -> None
+        """Restores soft-deleted share.
+
+        Operation will only be successful if used within the specified number of days
+        set in the delete retention policy.
+
+        :param str deleted_share_name:
+            Specifies the name of the deleted share to restore.
+            Service Version 2019-12-12 and later.
+        :param str deleted_share_version:
+            Specifies the version of the deleted share to restore.
+            Service Version 2019-12-12 and later.
+        :keyword int timeout:
+            The timeout parameter is expressed in seconds.
+        :rtype: None
```

Comment:
Could we also add the docstring tag to indicate which version of the SDK this method was introduced in?